### DISCUSSION TITLE
Difference between get() and prev?

### DISCUSSION THREAD

**Discussion by niaodan2b** [pmndrs/zustand#2151] [Q&A] [answered]
So instead of
```
set((prev) => ({ loading: !prev.loading }))
```
can I do
```
set({ loading: !get().loading })
```
Is there any difference? Any downsides to the latter?

**Comment by dai-shi** [comment, marked as answer]
They are basically identical.

https://github.com/pmndrs/zustand/blob/13439986a8cb036d80f86563955fcc891fdc2b2f/src/vanilla.ts#L71

https://github.com/pmndrs/zustand/blob/13439986a8cb036d80f86563955fcc891fdc2b2f/src/vanilla.ts#L83
